docs/devel/e2e-tests.md

```diff
@@ -232,6 +232,102 @@ stale permissions can cause problems.
 
   - `sudo iptables -F`, clear ip tables rules left by the kube-proxy.
 
+### Federation e2e tests
+
+By default, `e2e.go` provisions a single Kubernetes cluster, and any `Feature:Federation` ginkgo tests will be skipped.
+
+Federation e2e testing involve bringing up multiple "underlying" Kubernetes clusters,
+and deploying the federation control plane as a Kubernetes application on the underlying clusters.
+
+The federation e2e tests are still managed via `e2e.go`, but require some extra configuration items.
+
+#### Configuring federation e2e tests
+
+The following environment variables will enable federation e2e building, provisioning and testing.
+
+```sh
+$ export FEDERATION=true
+$ export E2E_ZONES="us-central1-a us-central1-b us-central1-f"
+```
+
+A Kubernetes cluster will be provisioned in each zone listed in `E2E_ZONES`. A zone can only appear once in the `E2E_ZONES` list.
+
+#### Image Push Repository
+
+Next, specify the docker repository where your ci images will be pushed.
+
+If you are using `KUBERNETES_PROVIDER=gce`, you can simply set your push repo base based on your project name, and the
+necessary repositories will be auto-created when you first push your container images:
+
+```sh
+$ export FEDERATION_PUSH_REPO_BASE="gcr.io/${GCE_PROJECT_NAME}"
+```
+
+Otherwise, if the provider is not gce, you'll be responsible for creating and managing access to the repositories manually.
+
+```sh
+$ export FEDERATION_PUSH_REPO_BASE="quay.io/colin_hom"
+```
+
+Given this example, the `federation-apiserver` container image will be pushed to the repository `quay.io/colin_hom/federation-apiserver`.
+
+The docker client on the machine running `e2e.go` must have push access for the following pre-existing repositories:
+
+* `${FEDERATION_PUSH_REPO_BASE}/federation-apiserver`
+* `${FEDERATION_PUSH_REPO_BASE}/federation-controller-manager`
+
+These repositories must allow public read access, as the e2e node docker daemons will not have any credentials.
```

Comment:
Its not clear that this is not true when user is using gce and his gcloud is configured correctly.